cmd/storage-class.go

```diff
@@ -115,68 +128,39 @@ func parseStorageClass(storageClassEnv string) (sc storageClass, err error) {
 	return sc, nil
 }
 
-// Validates the parity disks for Reduced Redundancy storage class
-func validateRRSParity(rrsParity, ssParity int) (err error) {
-	disks := len(globalEndpoints)
-	// disks < 4 means this is not a erasure coded setup and so storage class is not supported
-	if disks < 4 {
+// Validates the parity disks.
+func validateParity(ssParity, rrsParity int) (err error) {
+	if ssParity == 0 && rrsParity == 0 {
+		return nil
+	}
+
+	if !globalIsXL && !globalIsDistXL {
 		return fmt.Errorf("Setting storage class only allowed for erasure coding mode")
 	}
 
-	// Reduced redundancy storage class is not supported for 4 disks erasure coded setup.
-	if disks == 4 && rrsParity != 0 {
-		return fmt.Errorf("Reduced redundancy storage class not supported for " + strconv.Itoa(disks) + " disk setup")
+	// SS parity disks should be greater than or equal to minimumParityDisks. Parity below minimumParityDisks is not recommended.
+	if ssParity < minimumParityDisks && ssParity > 0 {
+		return fmt.Errorf("Standard storage class parity %d should be greater than or equal to %d", ssParity, minimumParityDisks)
 	}
 
 	// RRS parity disks should be greater than or equal to minimumParityDisks. Parity below minimumParityDisks is not recommended.
-	if rrsParity < minimumParityDisks {
-		return fmt.Errorf("Reduced redundancy storage class parity should be greater than or equal to " + strconv.Itoa(minimumParityDisks))
+	if rrsParity < minimumParityDisks && rrsParity > 0 {
+		return fmt.Errorf("Reduced redundancy storage class parity %d should be greater than or equal to %d", rrsParity, minimumParityDisks)
 	}
 
-	// Reduced redundancy implies lesser parity than standard storage class. So, RRS parity disks should be
-	// - less than N/2, if StorageClass parity is not set.
-	// - less than StorageClass Parity, if Storage class parity is set.
-	switch ssParity {
-	case 0:
-		if rrsParity >= disks/2 {
-			return fmt.Errorf("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(disks/2))
-		}
-	default:
-		if rrsParity >= ssParity {
-			return fmt.Errorf("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(ssParity))
-		}
+	if ssParity > len(globalEndpoints)/2 && ssParity > 0 {
```

Comment:
`ssParity > 0` is not needed here. `globalEndpoints` is never going to be 0 or 1, since we know its an erasure code backend.

Reply:
done